### PR TITLE
[Native] Add e2e cast int/float to decimal tests

### PR DESCRIPTION
**Pull request by rrando901**
## Description

This PR adds e2e tests for casting:
1. All integer types to decimals and decimals to all integer types.
2. REAL/DOUBLE to decimals and decimals to real/double.

### CODE REVIEW

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 1]
```diff
@@ -326,6 +326,27 @@ public void testCast()
         // Round-trip tests of casts for Json.
         assertQuery("SELECT cast(cast(name as JSON) as VARCHAR), cast(cast(size as JSON) as INTEGER), cast(cast(size + 0.01 as JSON) as DOUBLE), cast(cast(size > 5 as JSON) as BOOLEAN) FROM part");
         assertQuery("SELECT cast(cast(array[suppkey, nationkey] as JSON) as ARRAY(INTEGER)), cast(cast(map(array[name, address, phone], array[1.1, 2.2, 3.3]) as JSON) as MAP(VARCHAR(40), DOUBLE)), cast(cast(map(array[name], array[phone]) as JSON) as MAP(VARCHAR(25), JSON)), cast(cast(array[array[suppkey], array[nationkey]] as JSON) as ARRAY(JSON)) from supplier");
+
+        // Cast all integer types to decimal
+        assertQuery("SELECT CAST(linenumber_as_tinyint as DECIMAL(25, 0)) FROM lineitem");
```
Comment (karteekmurthys) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:331]: nit: Group the tests as short decimals and long decimals separated by comment lines.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 2]
```diff
@@ -326,6 +326,27 @@ public void testCast()
...
+        assertQuery("SELECT CAST(linenumber_as_smallint as DECIMAL(8, 4)) FROM lineitem");
+        assertQuery("SELECT CAST(CAST(linenumber as INTEGER) as DECIMAL(15, 6)) FROM lineitem");
+        assertQuery("SELECT CAST(nationkey as DECIMAL(21, 6)) FROM nation_partitioned");
+
+        // Cast decimal to all integer types
```
Comment (karteekmurthys) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:336]: Create a VALUES () list and have more than 1 value including nulls for casting to integers.
Again,  group short and long decimal test cases.

Refer other decimal tests in this file on how to create VALUES() list.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 3]
```diff
@@ -326,6 +326,27 @@ public void testCast()
...
+        assertQuery("SELECT CAST(DECIMAL'123.1234' as SMALLINT)");
+        assertQuery("SELECT CAST(DECIMAL'12345.1234' as INTEGER)");
+        assertQuery("SELECT CAST(DECIMAL'123456789.1234567' as BIGINT)");
+
+        // Cast double/float to decimal
+        // Casting  DOUBLE to DECIMAL and REAL to DECIMAL is not supported yet.
+        // assertQuery("SELECT CAST(discount as DECIMAL(8, 4)) FROM lineitem");
+        // assertQuery("SELECT CAST(discount_as_real as DECIMAL(4, 0)) FROM lineitem");
+
+        // Cast decimal to double/float.
```
Comment (karteekmurthys) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:347]: Same as above. Need to add more than 1 test values.

**Comment by karteekmurthys** [comment]
The CI build linux-presto-e2e-tests failed. Would you please take a look. Also, update the PR description. The dependent PR is closed.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 4]
```diff
@@ -390,6 +390,56 @@ public void testCast()
...
+
+        // Casting DOUBLE to DECIMAL and REAL to DECIMAL is not supported yet.
+
+        // Cast double/float to short decimal
+        // assertQuery("SELECT CAST(discount as DECIMAL(8, 4)) FROM lineitem");
+        // assertQuery("SELECT CAST(discount_as_real as DECIMAL(4, 0)) FROM lineitem");
+
+        // Cast double/float to long decimal
+        // assertQuery("SELECT CAST(discount as DECIMAL(21, 4)) FROM lineitem");
+        // assertQuery("SELECT CAST(discount_as_real as DECIMAL(19, 0)) FROM lineitem");
```
Comment (karteekmurthys) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:414]: Uncomment changes

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 5]
```diff
@@ -390,6 +390,56 @@ public void testCast()
...
+
+        // Cast short decimal to all possible integer types
+        assertQuery("SELECT CAST(c0 as TINYINT) FROM (VALUES (DECIMAL'1.23'), " +
```
Comment (karteekmurthys) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:417]: Remove "all possible"

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 6]
```diff
@@ -390,6 +390,56 @@ public void testCast()
...
+                "(NULL), (DECIMAL'1.'), (DECIMAL'0.0')) as l (c0)");
+        assertQuery("SELECT CAST(c0 as SMALLINT) FROM (VALUES (DECIMAL'123.04'), " +
+                "(NULL), (DECIMAL'32.760'), (DECIMAL'0.0')) as l (c0)");
+        assertQuery("SELECT CAST(c0 as INTEGER) FROM (VALUES (DECIMAL'12345.678'), " +
+                "(NULL), (DECIMAL'123456.7890'), (DECIMAL'1234567.0'), (DECIMAL'0.0')) as l (c0)");
+        assertQuery("SELECT CAST(c0 as BIGINT) FROM (VALUES (DECIMAL'1234567.89012134'), " +
+                "(NULL), (DECIMAL'12345678.901'), (DECIMAL'123456789.01214234')) as l (c0)");
+
+        // Cast long decimal to all possible integer types
```
Comment (karteekmurthys) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:426]: Same as above.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java (hunk 4 above)
Comment (rrando901) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:413]: Uncommenting these lines (as well as lines 408-409) results in failures in CI build linux-presto-e2e-tests.

`VeloxUserError:  Cast from DOUBLE to DECIMAL(8,4) is not supported cast((discount) as DECIMAL(8,4))
`

Comment (karteekmurthys) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:413]: Yes, please remove the changes. I forgot this is not implemented yet.

**Review by karteekmurthys — APPROVED** [review]
LGTM! Remove commented code. @aditi-pandit @majetideepak would you please review as well.

**Review by aditi-pandit — APPROVED** [review]

**Pull request merged**